afl-fuzz.c

```diff
@@ -1306,6 +1312,63 @@ static void update_bitmap_score(struct queue_entry* q) {
 
 }
 
+double rand_double() 
+{
+  // return random value in interval [0.0,1.0)
+  double rnd = (double)UR(RAND_MAX);
+  double max = (double)RAND_MAX;
+  return rnd / max;
+}
+
+// find the first element in array that is greater or equal target
+int first_greater_or_equal_element(double arr[], double target, int end)
+{
+    int start = 0;
```

Comment:
I think we can simplify this a bit. Here's a suggestion:

```C
int lo = 0;
int hi = end;
while (lo < hi) {
  int mid = lo + ((hi-lo)/2);
  if (target < arr[i]) {
    hi = mid;    
  } else {
    lo = mid + 1;
  }
}
return lo;
```

Reply:
Fixed